iOSDeviceManager/Utilities/DeviceUtils.m

```diff
@@ -100,52 +126,47 @@ + (NSString *)findDeviceIDByName:(NSString *)name {
-            NSString *otherSimNumber = [[otherSimDeviceName componentsSeparatedByCharactersInSet:nonDigitCharacterSet] componentsJoinedByString:@""];
-            if (simNumber.length == 0) {
-                return NSOrderedAscending;
-            }
-            if (otherSimNumber.length == 0) {
-                return NSOrderedSame;
-            }
-            if (fabs(simNumber.doubleValue - otherSimNumber.doubleValue) < EPSILON) {
-                // Handle things like 6S vs 6
-                NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:@".+\\d+[Ss]" options:0 error:nil];
-                NSUInteger simIsS = [regex numberOfMatchesInString:simDeviceName options:0 range:NSMakeRange(0, [simDeviceName length])];
-                NSUInteger otherSimIsS = [regex numberOfMatchesInString:otherSimDeviceName options:0 range:NSMakeRange(0, [otherSimDeviceName length])];
-
-                if (simIsS > 0 && otherSimIsS == 0) {
-                    return NSOrderedDescending;
-                }
-            } else if ((simNumber.doubleValue - otherSimNumber.doubleValue) > EPSILON) {
-                return NSOrderedDescending;
-            } else {
-                return NSOrderedAscending;
-            }
-        } else if ([simDeviceName containsString:@"iPhone"] && ![otherSimDeviceName containsString:@"iPhone"]) {
-            return NSOrderedDescending;
+    // step 1. define desired simulator model and runtime
+    int xcode_major;
+    int xcode_minor;
+    XcodeVersion(&xcode_major, &xcode_minor);
+    // TODO: runtime versionwill be used in iOS version comparision
+    // int major=xcode_major+2;
+    // int minor=xcode_minor;
+    
+    NSString *defaultModel;
+    
+    if (xcode_major == 10) {
+        defaultModel=@"XS";
+    }else{
+        defaultModel=[NSString stringWithFormat:@"%d", xcode_major-1];
+    }
+    
+    // step 2. find FBSimulator with the desired simulators
+    // Re explanation:
+    //   iPhone\\s+  - skip anything before model name
+    //   (\\d+|XS)   - pickup either XS or one of 4..8 model and save into capture group #1
+    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:@"iPhone\\s+(\\d+|XS)" options:0 error:nil];
+    
+    // while we do not have iOS information will assign each matched simulator
+    // to "def" variable. The last one will be returned
+    FBSimulator *def = nil;
+    for (FBSimulator *simulator in simulators) {
+        NSString *simName = [simulator name];
+        NSArray *matches = [regex matchesInString:simName options:0 range:NSMakeRange(0, [simName length])];
+        if (matches==nil || matches.count == 0) continue;
+
+        // rangeAtIndex:0 - the whole match
+        // rangeAtIndex:0 - the first captured group #1
+        NSString *model = [simName substringWithRange:[(NSTextCheckingResult*)matches[0] rangeAtIndex:1]];
+        if ([model isEqualToString:defaultModel]) {
+            def=simulator;
```

Comment:
I would return early here:

```
return simulator;
```

there is no need to continue to iterate.

Reply:
The idea is to iterate till the last entry to get the recent iOS version.